test/filt_stream.jl

```diff
@@ -102,8 +102,9 @@ function test_singlerate(h::AbstractVector{T}, x::AbstractVector) where T
     @printfifinteractive( "\n\tfilt filt. Piecewise for first %d inputs\n\t\t", length(x1) )
     reset!(myfilt)
     @timeifinteractive begin
-        for i in 1:length(x1)
-            y1[i] = filt(myfilt, x1[i:i])[1]
+        one_buffer = similar(x1, 1)
+        for (i, p) in enumerate(x1)
+            y1[i] = filt(myfilt, setindex!(one_buffer, p, 1))[1]
```

Comment:
I'm not sure these `@timeinteractive` should be taken seriously at all (I never look at them), so this optimization might be nothing more than an unnecessary complication. But even if care about the allocations here, wouldn't
```suggestion
        for i in eachindex(x1)
            y1[i] = filt(myfilt, @view(x[i:i]))[1]
```
cut it? (And likewise below.)

Reply:
Ah, that's unfortunate. Rework after fixing #605, then :smile: